### PR TITLE
Add support for Alpine morph and custom morphers

### PR DESCRIPTION
**Pull request by imankulov**
To replace morphdom wih Alpine morph:

Add to your base template:

    <script defer src="https://unpkg.com/@alpinejs/morph@3.x.x/dist/cdn.min.js"></script>
    <script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>

Add to your settings:

    UNICORN = {
        "MORPHER": "alpine",
    }

Ref: https://alpinejs.dev/plugins/morph

### CODE REVIEW

**Comment by imankulov** [comment]
I created a demo project and recorded a video to show how it works:

- [Video](https://www.loom.com/share/5593106544384aa2acbf57701f5f4e6b)
- [Sample project. Branch experiment/alpine-morph](https://github.com/imankulov/django-unicorn-demo/tree/experiment/alpine-morph)

**Comment by imankulov** [comment]
@adamghill, when you have time, could you please take a look at 5f240f5f02e4fdd6cca65d078f1283ba2321b535 where I document the future state of custom morphers and let me know if you have any feedback? Thanks!

**Comment by imankulov** [comment]
@adamghill, everything except tests is ready for the review.

If you're OK with my work so far, can you please pick it up from here. Alternatively, can you please help me understand which tests should I add and modify to make it work? Thanks!

File: django_unicorn/static/unicorn/js/morphers/alpine.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      throw Error(`
+Alpine morpher requires Alpine to be loaded. Add Alpine and Alpine Morph to your page.
+See https://www.django-unicorn.com/docs/custom-morphers/#alpine for more information.
+`);
+    }
+    this.options = options;
+  }
+
+  morph(dom, htmlElement) {
+    if (htmlElement) {
```
Comment (imankulov) [inline django_unicorn/static/unicorn/js/morphers/alpine.js:14]: @adamghill, frankly, I couldn't understand why morph is sometimes called with an empty string as "htmlElement".

**Comment by adamghill** [comment]
> @adamghill, when you have time, could you please take a look at [5f240f5](https://github.com/adamghill/django-unicorn/commit/5f240f5f02e4fdd6cca65d078f1283ba2321b535)

This looks like a good start! I'll probably tweak it a bit when I get a change, but thanks for writing up all of this.

I rebased `main` into this branch and have started working on the tests. I will continue to fix up the JavaScript tests and make sure things look good to get this into a mergeable state for the next release.

**Comment by imankulov** [comment]
Thanks a lot for picking up where I left off, @adamghill! I'm looking forward to having these changes merged. 🙌

**Comment by imankulov** [comment]
**Here is the summary of the last commits that I made yesterday and today.**

I was almost ready to use "alpine-morph" in production, but then I found that alpine.min.js, which is used when DEBUG is False, is not up-to-date with the source code.

Additionally, the getMorpher() function needs to be moved to a Unicorn namespace so that it can be accessed from the minified version.

Finally, I discovered that Alpine and Alpine morph will be loaded after our script (as they are loaded with <script defer>), which causes the Alpine morpher constructor to throw an "Alpine not found" exception.

To address these changes:

- I moved getMorpher() to unicorn.js so that external scripts can call it as "Unicorn.getMorpher()".
- I moved the validation for the existence of Alpine and Alpine morph from the constructor to morph().
- I recompiled unicorn.min.js.

In addition to this, I created a separate PR https://github.com/adamghill/django-unicorn/pull/604 to ensure that unicorn.min.js is always up to date with the source code.

**Comment by adamghill** [comment]
@imankulov Thanks for all your work for this PR. This got merged into `main` and should be available in 0.56.0. 🎉

**Comment by adamghill** [comment]
@imankulov I ended up releasing 0.56.1 with another fix for the Alpine morpher. Can you try it out and let me know if you run into any problems?

**Comment by imankulov** [comment]
@adamghill, I have tested version 0.56.1 with my codebase that uses the Alpine morpher, and everything appears to be functioning correctly 🎉 

Thank you very much for completing and merging the PR.

**Pull request closed without merging**